src/rebuild/README.md

```diff
@@ -1,275 +1,150 @@
-
-<a id="10.5.6.1"></a>
-#### Consistency
-
-In order to achieve the goal described above, the DSR adopts a simple protocol:
-
-- A DSR I/O request can complete only if underlying I/O requests against all RDG members succeeded.
-
-More particular, this protocol is for the client side data protection:
-
-- A DSR object update can complete only if updates of all the object shards of the RDG have successfully completed.
-- If any of these updates failed, the DSR client should infinitely retry until it succeeds, or there is a pool map change which shows the target failed. In the second case, the DSR client should switch to the new pool map, and send the update to the new destination, which is the rebuild target of this object.
-
-The <a href="#f10.19">figure</a> below shows an example of this client protocol.
-
-Also worth mentioning is that this protocol is based on assumption that Raft leader can get the RAS notification and broadcast the failure to all targets in a reasonable short time. Otherwise, the DSR I/O request can hang for long time.
-
-<a id="f10.19"></a>
-**DSR client protocol for online rebuild**
-![../../doc/graph/Fig_060.png](../../doc/graph/Fig_060.png "DSR client protocol for online rebuild")
-
-For the server side data protection, there are many researches that introduce various protocols to provide data consistency, however, given the epoch and I/O model of DAOS, most of them are unnecessary:
-
-- Server side erasure coding can only happen to already committed epoch, so there is no concurrent write for them.
-- There is no client cache and distributed lock in DAOS, so server side replication can also be greatly simplified, for example, the primary replica can even use the similar protocol as the client replication to guarantee data consistency for online rebuild.
-
-<a id="10.5.6.2"></a>
-#### Data Fence
-
-If an application keeps writing to the object that is in rebuilding, and the writing throughput is equal to or higher than the rebuild throughput, then the rebuilding process needs a fence for data, so it can complete the rebuild when all data have reached at the destination shard, otherwise the rebuilding process could be endless.
-
-Because all I/O requests should carry the pool map version, the data fence can be the pool map version when an extent or KV was updated. The data fence should be stored as metadata of an extent or KV in VOS. The rebuilding process can skip those data fenced by the new pool map version.
-
-The data fence could also be used to resolve the conflict between application writes and rebuilding writes: application writes should overwrite rebuilding writes in the VOS data tree even they are in the same epoch.
-
-<a id="10.5.7"></a>
-### Multi-Failure
```

Comment:
is there any reason to remove the whole "multi-failure" section, do you have plan to add it back?

Reply:
Oh, I described it in "Multiple pool and targets rebuild". But only describe "what" will happen, not "why" this happen like HLD.  since this is README anyway.

Reply:
The goal of this is moving HLD to readme so the design can stay together with source code...